pkg/doc/presexch/definition.go

```diff
@@ -663,25 +663,55 @@ func createNewCredential(constraints *Constraints, src, limitedCred []byte,
 		return verifiable.ParseUnverifiedCredential(limitedCred, opts...)
 	}
 
-	var err error
-
-	for _, path := range explicitPaths {
-		limitedCred, err = sjson.SetBytes(limitedCred, path, true)
-		if err != nil {
-			return nil, err
-		}
+	limitedCred, err := enhanceRevealDoc(explicitPaths, limitedCred, src)
+	if err != nil {
+		return nil, err
 	}
 
 	var doc map[string]interface{}
 	if err := json.Unmarshal(limitedCred, &doc); err != nil {
 		return nil, err
 	}
 
-	doc["@explicit"] = true
-
 	return credential.GenerateBBSSelectiveDisclosure(doc, []byte(uuid.New().String()), opts...)
 }
 
+func enhanceRevealDoc(explicitPaths map[string]bool, limitedCred, vcBytes []byte) ([]byte, error) {
+	var err error
+
+	for path := range explicitPaths {
```

Comment:
We are missing the part when the path is few levels deep.
Let's say we have path `subject.user.address`.
The current implementation sets values only for the full path.
e.g 
```
subject.user.address.@explicit
subject.user.address.type
subject.user.address.@context
```

In that case, we are missing everything between.
e.g
```
# root
@explicit
type
@context

# first level
subject.@explicit
subject.@type
subject.@context

# second level
subject.user.@explicit
subject.user.@type
subject.user.@context
```

Reply:
Fixed!